src/dns_cache_handler.erl

```diff
@@ -0,0 +1,176 @@
+%%%-------------------------------------------------------------------
+%%% @author dmitry.pravosudov
+%%% @copyright (C) 2019, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 20. Dec 2019 11:48
+%%%-------------------------------------------------------------------
+-module(dns_cache_handler).
+-author("dmitry.pravosudov").
+
+-behaviour(gen_server).
+
+%% API
+-export([
+    start_link/0,
+    get_cached_dns/1
+]).
+
+%% gen_server callbacks
+-export([init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2,
+    code_change/3]).
+
+-define(SERVER, ?MODULE).
+-define(TAB, ddb_host_cache).
+-define(ETSTTL, 86400).
+
+-record(state, {
+}).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+-spec(get_cached_dns(Host :: string()) ->
+    {A :: integer(), B :: integer(), C :: integer(), D :: integer()} | string()).
+get_cached_dns(Host0) ->
+    {_, Domains} = application:get_env(lhttpc, domains_to_cache),
+    {_, {Units, Divider}} = application:get_env(lhttpc, dns_cache_opts),
+    
+    Key = floor(erlang:system_time(Units) / Divider),
```

Comment:
This has an issue 

The Key does not have the Host into account this means that you might hit in the same second a slot that does not relate with the IP of the Host you are using. 

Imagine you are caching "dynamodb.us-west-2.amazonaws.com" and "sqs.us-west-2.amazonaws.com" with the current implementation if you hit the same slot in the same second  all the request after the first (i.e dynamodb host) will resolve to the  IP that might not be the one for the mentioned host (i.e. sqs host).

You should add the host to the key otherwise we might face weird issues, because the  cached IP does not belong to the Host.

Reply:
Yes, I found this too. I fixed it.